tests/e2e/ctl_v3_member_test.go

```diff
@@ -304,3 +312,230 @@ func TestRemoveNonExistingMember(t *testing.T) {
 	// Ensure that membership is properly bootstrapped.
 	assert.NoError(t, epc.Restart(ctx))
 }
+
+// TestClusterUpgradeWithPromotedLearner tests whether etcd can automatically
+// fix the issue caused by https://github.com/etcd-io/etcd/issues/19557
+// when upgrading from 3.5 to 3.6.
+func TestClusterUpgradeWithPromotedLearner(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		snapshotCount         uint64
+		writeToV3StoreSuccess bool
+	}{
+		{
+			name:          "create snapshot after learner promotion which is not saved to v3store",
+			snapshotCount: 10,
+		},
+		{
+			name:          "no snapshot and learner promotion is not saved to v3store",
+			snapshotCount: 0,
+		},
+		{
+			name:                  "no snapshot and learner promotion is saved to v3store",
+			snapshotCount:         0,
```

Comment:
Why not test with `shapshotCount: 10` and `writeToV3StoreSuccess: true,`

Reply:
We were even planning to remove `writeToV3StoreSuccess`, the only reason is to test the normal case, ensure it goes into "`ignore already promoted member`" branch.

Reply:
There is nothing to expect when `shapshotCount: 10` and `writeToV3StoreSuccess: true`.

Reply:
Aaa, ok I mistook the conditions. Thought it's reversed. Now it makes sense. Thanks.